### PR TITLE
Verbose config extra fact

### LINKED ISSUE #20: facts subset config is allways "changed"
##### SUMMARY

Every time 'community.routeros.facts' subset 'config' returns data it's status is 'not changed' but it is false.
Returned data are different on every run of 'community.routeros.facts' even configuration on device is not changed:
1. there is date at the beggining of the file
2. with "/export verbose" there are exported unconfigured services too and for example unconfigured '/interface ovpn-server server' contains random generated MAC address (tested on RouterBOARD 750G r3)

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

'community.routeros.facts' subset 'config'

##### ANSIBLE VERSION

```paste below

```

##### CONFIGURATION

```paste below
# ansible --version
ansible 2.10.6
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/root/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.8/dist-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 3.8.5 (default, Jul 28 2020, 12:59:40) [GCC 9.3.0]
```

##### OS / ENVIRONMENT

Ubuntu 20.04
RouterBOARD 750G r3
RouterOS 6.40.3

##### STEPS TO REPRODUCE

setup your inventory to an Mikrotik device (tested on RouterBOARD 750G r3)

```yaml
---
- name: Mikrotik config1
  community.routeros.facts:
    gather_subset: config

- set_fact: cfg1="{{ ansible_net_config }}" 
- set_fact: cfg3="{{ ansible_net_config }}" 

- name: Mikrotik config2
  community.routeros.facts:
    gather_subset: config

- set_fact: cfg2="{{ ansible_net_config }}" 

- ansible.builtin.debug: 
    msg: "cfg1 is different from cfg2"
  when: cfg1 != cfg2

- ansible.builtin.debug: 
    msg: "cfg1 is not different from cfg1"
  when: cfg1 == cfg1

- ansible.builtin.debug: 
    msg: "cfg1 is not different from cfg3"
  when: cfg1 == cfg3
```

##### EXPECTED RESULTS

cfg1 should be the same cfg2 because device configuration did not changed

##### ACTUAL RESULTS

cfg1 and cfg2 are different.
it makes very difficult to version those cfg files

### PR DESCRIPTION
**Pull request by qaxi**
##### SUMMARY

Every time 'community.routeros.facts' subset 'config' returns data it's status is 'not changed' in perspective of device it is OK, but ...
Returned data are different on every run of 'community.routeros.facts' even configuration on device is not changed

This adds fact _ansible_net_config_nonverbose_ which contain idempotent config (not changing on every _community.routeros.facts_ execution).

Fixes #20 

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

'community.routeros.facts' subset 'config'

### CODE REVIEW

**Comment by qaxi** [comment]
Is there any chance to run tests again?

The error message is
"ERROR! Error when getting available collection versions for ansible.netcommon from default (https://galaxy.ansible.com/api/) (HTTP Code: 520, Message: Origin Error Code: Unknown)"

Tests ran OK on https://github.com/qaxi/community.routeros/actions/runs/686061241

File: changelogs/fragments/1.2.0.yml [hunk 1]
```diff
@@ -0,0 +1 @@
+release_summary: This release add nonverbose idempotent config in facts
```
Comment (felixfontein) [inline changelogs/fragments/1.2.0.yml:1]: Please remove this file.

File: changelogs/fragments/23-indempotent_config.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - fact - add fact _ansible_net_config_nonverbose_ to get indempotent config (no date, no verbose)
```
Comment (felixfontein) [inline changelogs/fragments/23-indempotent_config.yml:2]:
```suggestion
  - fact - add fact ``ansible_net_config_nonverbose`` to get indempotent config (no date, no verbose) (https://github.com/ansible-collections/community.routeros/pull/23).
```

File: plugins/modules/facts.py [hunk 3]
```diff
@@ -280,14 +280,26 @@ def to_megabytes(self, data):
...
+
         data = self.responses[0]
         if data:
             self.facts['config'] = data
 
+        data = self.responses[1]
+        if data:
+            # remove datetime
+            data = re.sub(self.RM_DATE_RE, r'# RouterOS', data)
+            self.facts['config_nonverbose'] = data
```
Comment (felixfontein) [inline plugins/modules/facts.py:301]: You need to document this in `RETURNS`.

File: plugins/modules/facts.py [hunk 4]
```diff
@@ -280,14 +280,26 @@ def to_megabytes(self, data):
 
 class Config(FactsBase):
 
-    COMMANDS = ['/export verbose']
+    COMMANDS = [
+        '/export verbose',
+        '/export'
```
Comment (felixfontein) [inline plugins/modules/facts.py:285]:
```suggestion
        '/export',
```

File: tests/unit/plugins/modules/test_facts.py [hunk 5]
```diff
@@ -90,10 +90,15 @@ def test_facts_hardware(self):
     def test_facts_config(self):
         set_module_args(dict(gather_subset='config'))
         result = self.execute_module()
+        print(result['ansible_facts'])
```
Comment (felixfontein) [inline tests/unit/plugins/modules/test_facts.py:93]:
```suggestion
```

**Comment by felixfontein** [comment]
I assume your intention is that one of #22 and this PR should get merged?

**Comment by qaxi** [comment]
Yes, I dont know which one suits you better ...

File: plugins/modules/facts.py [hunk 6]
```diff
@@ -110,6 +110,11 @@
       returned: when config is configured
       type: str
 
+    ansible_net_config_nonverbose:
+      description: The current active config from the device in minimal form
+      returned: when config is configured
+      type: str
```
Comment (felixfontein) [inline plugins/modules/facts.py:116]:
```suggestion
      type: str
      version_added: 1.2.0
```

File: plugins/modules/facts.py [hunk 7]
```diff
@@ -110,6 +110,11 @@
       returned: when config is configured
       type: str
 
+    ansible_net_config_nonverbose:
+      description: The current active config from the device in minimal form
```
Comment (felixfontein) [inline plugins/modules/facts.py:114]:
```suggestion
      description:
          - The current active config from the device in minimal form.
          - This value is idempotent in the sense that if the facts module is run twice and the device's config
            was not changed between the runs, the value is identical. This is achieved by running C(/export)
            and stripping the timestamp from the comment in the first line.
```

File: changelogs/fragments/23-indempotent_config.yml [hunk 8]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - fact - add fact ``ansible_net_config_nonverbose`` to get indempotent config (no date, no verbose) (https://github.com/ansible-collections/community.routeros/pull/23).
```
Comment (felixfontein) [inline changelogs/fragments/23-indempotent_config.yml:2]:
```suggestion
  - fact - add fact ``ansible_net_config_nonverbose`` to get idempotent config (no date, no verbose) (https://github.com/ansible-collections/community.routeros/pull/23).
```

**Comment by felixfontein** [comment]
With that typo fixed, I think this is ready for merging! @heuels what do you think?

**Review by heuels — APPROVED** [review]

**Comment by heuels** [comment]
@felixfontein, yes, LGTM!

**Comment by heuels** [comment]
@qaxi, I just noticed that you also have the typo in the changelog filename 👀

**Comment by qaxi** [comment]
Damn dyslexia ... thank you.

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@qaxi thanks for this contribution!
@heuels thanks for reviewing!

**Comment by felixfontein** [comment]
@heuels should I do a 1.2.0 release soon (today/tomorrow so it can get included in the next Ansible 3.x.0 release)? Or should we wait a bit more if more things come up?

**Pull request merged**